### ISSUE TITLE
In the training set, the content of the picture is a house far away from the photographer, so will it cause a big error to use a close-range picture of the house to predict in test time?

### ISSUE THREAD

**Issue by wpumain** [gmberton/CosPlace#18]

**Comment by gmberton** [comment]
The CosPlace model automatically learns to pay more attention to nearby static objects, while paying less attention to far away objects (e.g. a tall building in the distance) or moving objects (cars and pedestrians).
This is true also for previous methods, such as NetVLAD.

**Comment by wpumain** [comment]
Thank you for your answer.But how to understand this sentence  【The CosPlace model automatically learns to pay more attention to nearby static objects, while paying less attention to far away objects 】 in terms of network model structure or mathematics?

**Comment by gmberton** [comment]
I don't really understand what you mean. All I want to say is that the model's descriptors depend mostly on things that can help to understand the camera position, which usually are buildings not too far away.

**Comment by wpumain** [comment]
How does this model do this 【The CosPlace model automatically learns to pay more attention to nearby static objects, while paying less attention to far away objects 】？

**Comment by gmberton** [comment]
That's what the loss is teaching the model. To group together (close in features space) images with the same orientation within a 10x10 meter cell, and keep their features far from other images (belonging to other cells), the only way is to focus on nearby static objects

**Comment by wpumain** [comment]
Wow, I finally understand.The features of the objects close to each other have strong correlation。Does that mean that when we make training data sets, these images should be close-range and static  photos?

**Comment by gmberton** [comment]
When we collect datasets, the images will contain some close-range static objects for sure. It is very rare for a photo taken outdoor to have absolutely no nearby static objects.

**Comment by wpumain** [comment]
Think you for your help
